### ISSUE TITLE
[hackc] Can't create closure in anonymous classes

### ISSUE THREAD

**Issue by Norbyte** [facebook/hhvm#8243]
### HHVM Version

HipHop VM 3.27.0 (dbg)
Compiler: tags/HHVM-3.27.0-0-ga0991a6923462ba8b66244f836d4cbd1c616467a
Repo schema: 2ee5d5c5e299ea0fff5870859d33ca2711fb7c22

### Operating System and Version

CentOS Linux release 7.5.1804 (Core)

### Standalone code, or other way to reproduce the problem

```
<?php
function x()
{
    $a = new class {
        public function func()
        {
            $closure = function () {
                echo 'ok?';
            };

            $closure();
        }
    };

};

x();
```

In HHVM 3.27 I can no longer create closures inside anonymous classes. In HHVM 3.23 and vanilla PHP 7.1 this works fine, i.e. it outputs "ok?".

### Actual result

Fatal error: Uncaught Error: Can't create a Closure directly in /home/n/hhvm-test/hhtest.php:14
Stack trace:
#0 (): trigger_error()
#1 /home/n/hhvm-test/hhtest.php(14): Closure->__construct()
#2 /home/n/hhvm-test/hhtest.php(18): x()
#3 {main}

### Expected result

"ok?"

**Comment by fredemmott** [comment]
Thanks for the report; fb internal reference: t30919578

**Comment by fredemmott** [comment]
Fixed in f82258044f27dc226ee203412d3d9f8d72bd1cc8
